protocol/0045-DSRC-data_sourcing.md

```diff
@@ -246,3 +246,6 @@ Vega should reject any data source tx that is not explicitly required, so this w
 1. After trading termination has been triggered the trading terminated data source is no longer active (assuming it is not used anywhere else) and data from that source is no longer processed (<a name="0045-DSRC-019" href="#0045-DSRC-019">0045-DSRC-019</a>)
 1. After settlement data has been received and the market has settled, the settlement data source is no longer active (assuming it is not used anywhere else) and data from that source is no longer processed (<a name="0045-DSRC-020" href="#0045-DSRC-020">0045-DSRC-020</a>)
 1. Where the same oracle definition is used for trading terminated and settlement, and data has been received. Trading is terminated and the market is settled with the same data event (a single message does both). The data source is no longer active (assuming it is not used anywhere else) and data from that source is no longer processed. (<a name="0045-DSRC-021" href="#0045-DSRC-021">0045-DSRC-021</a>)
+1. It should be possible to update the market termination spec from external to internal and prove that the original termination key and filter no longer supports termination and that the updated market termination key and filter supports termination. This should apply to filter key types Timestamp/Boolean and Integer. (<a name="0045-DSRC-022" href="#0045-DSRC-022">0045-DSRC-022</a>)
+1. It should be possible to update the market termination spec from internal to external and prove that the original termination key and filter no longer supports termination and that the updated market termination key and filter supports termination. This should apply to filter key types Timestamp/Boolean and Integer. (<a name="0045-DSRC-023" href="#0045-DSRC-023">0045-DSRC-023</a>)
+1. It should not be possible to update the market settlement spec from external to internal type (<a name="0045-DSRC-024" href="#0045-DSRC-024">0045-DSRC-024</a>)
```

Comment:
What does this mean exactly?

Reply:
@davidsiska-vega , these conditions mean that 
1. We should be able to update a market termination from external to internal
2. We should be able to update a market termination from internal to external
3. Market settlement is only external
After updating the markets, the old conditions should not work.

In my opinion the changes are relevant, approving now.

Reply:
I agree with "0045-DSRC-022" and "0045-DSRC-023". 

I don't uderstand "0045-DSRC-024". Why should I not be allowed to change market settlement spec from external to internal?

Reply:
Because market settlement can be only external.

Reply:
@davidsiska-vega , currently the internal data source is implemented only for time terminated type, please refer to this:
https://github.com/vegaprotocol/specs/blob/master/protocol/0048-DSRI-data_source_internal.md#13-vega-time-changed

In the current core implementation this type does not contain anything except a timestamp and the builtin key name for it.

Reply:
This doesn't mean we need to support [internal value sources](https://github.com/vegaprotocol/specs/blob/master/protocol/0048-DSRI-data_source_internal.md#11-value), etc. at this time, to switch to, just that if there is a valid internal data source for a given consumer, we should be able to switch to/from it.